WcaOnRails/app/webpacker/components/CompetitionForm/VenueMap.js

```diff
@@ -0,0 +1,130 @@
+/* eslint-disable jsx-a11y/click-events-have-key-events */
+/* eslint-disable jsx-a11y/no-static-element-interactions */
+import React, { useRef, useState, useEffect } from 'react';
+import {
+  Circle,
+  Map,
+  Marker,
+  TileLayer,
+  ZoomControl,
+  useLeaflet,
+} from 'react-leaflet';
+import { GeoSearchControl as SearchControl, OpenStreetMapProvider } from 'leaflet-geosearch';
+import { userTileProvider } from '../../lib/leaflet-wca/providers';
+import { blueMarker } from '../../lib/leaflet-wca/markers';
+
+// Copied from lib/leaflet-wca/index.js which had nothing exported.
+function roundToMicrodegrees(toRound) {
+  const val = toRound || 0;
+  // To prevent are you sure? from firing even when nothing has changed,
+  // explicitly round coordinates to an integer number of microdegrees.
+  return Math.trunc(parseFloat(val) * 1e6) / 1e6;
+}
+
+function DraggableMarker({ latData, longData }) {
+  const position = { lat: latData.value, lng: longData.value };
+  const markerRef = useRef(null);
```

Comment:
I'm not too familiar with `useRef`, could you explain why it's needed here?

Reply:
Im not 100% certain, as I was followng the react-leaflet docs, but I think it's to do with how its handling storing the state of the marker and using it in the dragend callback

Reply:
You would use `useRef` whenever you need to reference the element. Either in an (antipattern) way to trigger a function on it or access data in a backwards way.

The more standard practice would be to retrieve this data from an event object.

from looking at source code, `onDragend` should pass in an "event" prop. Can you look at it to see if you can access the position from there?